iterating-over-data/practice-iterating-over-data.md

```diff
@@ -0,0 +1,465 @@
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x, y in my_profile.items()`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 6 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: NkbCNAWCF
+* title: Match the Output
+
+##### !question
+
+Given a function named `search_dictionary` that takes two parameters: `dict` (a dictionary) and `check` (a string), check if the key `check` is found in `dict`, and then return the value at that key.
+
+For example, given: `{"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}` and `has_pets`, return `True`.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+
+def search_dictionary(dict,check):
+  pass
+
+```
+
+##### !end-placeholder
+
+##### !tests
+```python
+
+import unittest
+from main import *
+
+dict = {"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}
+class TestPython1(unittest.TestCase):
+def test_one(self):
+  self.assertEqual(search_dictionary(dict, "has_dogs"))
+  self.assertEqual(search_dictionary(dict, "icecream_flavor"),"chocolate")
+
+```
+##### !end-tests
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+Two examples of working implementations:
+
+``` python
+
+def search_dictionary(dict,check):
+  for i in dict:
+    if i == check:
+      return dict[i]
+
+```
+``` python
+
+def search_dictionary(dict,check):
+  for key, value in dict.items():
+    if key == check:
+      return value
+
+```
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 7 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: Ny72-sS0Y
+* title: Match the Output, Cont.
+
+##### !question
+
+Given the function `search_dictionary` created above, let's add to it. If function runs through the `dict` and find no `check` key, then return `False`.
+
+For example, given: `{"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}` and `has_dogs`, return `False`.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+
+def search_dictionary(dict,check):
+  # add your code here from the previous challenge,
+  # and then add to it
+  pass
+
+```
+
+##### !end-placeholder
+
+##### !tests
+```python
+
+import unittest
+from main import *
+
+dict = {"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}
+class TestPython1(unittest.TestCase):
+def test_one(self):
+  self.assertEqual(search_dictionary(dict, "has_dogs"),False)
+  self.assertEqual(search_dictionary(dict, "icecream_flavor"),"chocolate")
```

Comment:
Why is `dict` defined outside of the unit test? It's totally appropriate to make it a variable local to the test itself

```suggestion
class TestPython1(unittest.TestCase):
    def test_one(self):
        dict = {"icecream_flavor": "chocolate", "lucky_nums": [7, 18, 34], has_pets: True}
        self.assertEqual(search_dictionary(dict, "has_dogs"),False)
        self.assertEqual(search_dictionary(dict, "icecream_flavor"),"chocolate")
```